Widgets.cpp

```diff
@@ -599,11 +599,17 @@ namespace ImGuiEx {
 			column.IsEnabledNextFrame = !column.IsEnabled;
 	}
 
-	void WindowReposition(Position position, const ImVec2& cornerVector, CornerPosition cornerPosition, ImGuiID fromWindowID,
+	bool WindowReposition(ImGuiWindow* window, Position position, const ImVec2& cornerVector, CornerPosition cornerPosition, ImGuiID fromWindowID,
 	                      CornerPosition anchorPanelCornerPosition, CornerPosition selfPanelCornerPosition) {
-		const ImVec2& windowSize = ImGui::GetWindowSize();
+		if (window == nullptr) {
+			window = ImGui::GetCurrentWindowRead();
```

Comment:
We call `SetWindowPos` on this window. So this should be should not get a read-only window.

Reply:
It's a misnomer in ImGui, this is what the imgui function looks like:
```
void ImGui::SetWindowPos(const ImVec2& pos, ImGuiCond cond)
{
    ImGuiWindow* window = GetCurrentWindowRead();
    SetWindowPos(window, pos, cond);
}
```

And the definition goes like
```
    bool                    WriteAccessed;                      // Set to true when any widget access the current window
```

Reply:
```
    inline    ImGuiWindow*  GetCurrentWindowRead()      { ImGuiContext& g = *GImGui; return g.CurrentWindow; }
    inline    ImGuiWindow*  GetCurrentWindow()          { ImGuiContext& g = *GImGui; g.CurrentWindow->WriteAccessed = true; return g.CurrentWindow; }
```
ok, ImGui, sometimes you're weird^^